common/src/main/scala/org/neo4j/spark/service/Neo4jQueryService.scala

```diff
@@ -276,31 +283,37 @@ class Neo4jQueryReadStrategy(filters: Array[Filter] = Array.empty[Filter],
       case "*" => Asterisk.INSTANCE
       case name => {
         val cleanedName = name.removeAlias()
-        aggregateColumns
-          .filter(_.toString == name)
-          .headOption
-          .map(ag => ag match {
+        aggregateColumns.find(_.toString == name)
+          .map {
             case count: Count => {
-              val prop = entity.property(count.column().describe().unquote().removeAlias())
+              val col = count.column().describe().unquote().removeAlias()
+              val prop = propertyOrSymbolicName(col)
```

Comment:
what does this do?

Reply:
```scala
    def propertyOrSymbolicName(col: String) = {
      if (entity != null) entity.property(col) else Cypher.name(col)
    }
```

Reply:
I did it because I tested the aggregation function with GDS and I needed to refactor/improve the code